frontend/javascripts/oxalis/model/reducers/volumetracing_reducer_helpers.js

```diff
@@ -51,12 +51,12 @@ export function createCellReducer(state: OxalisState, volumeTracing: VolumeTraci
     // cellId 0 means there is no annotation, so there must not be a cell with id 0
     return state;
   }
-  let newMaxCellId = volumeTracing.maxCellId;
+
+  // The maxCellId is only updated if a voxel using that id was annotated. Therefore, it can happen
+  // that the activeCellId is larger than the maxCellId. Choose the larger of the two ids and increase it by one.
+  const { activeCellId, maxCellId } = volumeTracing;
   if (id == null) {
-    newMaxCellId++;
-    id = newMaxCellId;
-  } else {
-    newMaxCellId = Math.max(id, newMaxCellId);
```

Comment:
This line change raised some questions for me which is why I digged a bit through the whole `cells` code. I have some observations which I cannot really get straight in my head:

- apparently, we have an entire `cells` mapping in the store (I wasn't really aware of that :see_no_evil:). however, this seems to be mostly unused (it's only maintained as far as I can see). for each cell, this mapping only stores its id, so there is no real valuable information there. BUT: we could probably easily build upon that (e.g., to add a position to each cell) which would be quite cool at some point.
- right now, a new cell is created as soon as its ID is made the `activeCellId`. in my opinion, this is a bit conflicting the new semantics of this PR which only considers a new cell id once it was also labeled. I think it would be more intuitive if the maximum cell ID would always equal the maximum key id in `cells`. however, it's probably not really crucial to have this constraint (and also might be hard to implement since the creation should ideally happen before something is labeled).
- passing an explicit cell id to `createCell` seems a bit weird to me. as far as I can see, there are only two places where this is used: in `setActiveCellReducer` and in the tests. For `setActiveCellReducer`, the use case makes sense. however, the `cellId` parameter in the redux `createCell`-action should probably never really be used. so, maybe we should just remove this from the action?
- if a not-null id parameter is only used in `setActiveCellReducer`, the following point is probably irrelevant. however, considered in isolation, this specific line change seems a bit dangerous to me. if this function is called with a cell id for which there is already an object in the store, the function would create a new `cell` object and replace it. to avoid this, this reducer could simply check whether the cell already exists and then do nothing (would be a bit redundant to what `setActiveCellReducer` does right now, but theoretically other callers could be added in the future).

Sorry for this wall of text. I just wanted to share what went through my head. We could also have a call about this, in case it's easier. My tldr-suggestions would be probably be: remove the `cellId` parameter from the `createCell` action & don't do anything in `createCellReducer` if the id already exists. Creating the cell object not-lazily is probably alright, as that structure is not stored on the server anyway (at least for now).

Reply:
> Sorry for this wall of text. I just wanted to share what went through my head. We could also have a call about this, in case it's easier. My tldr-suggestions would be probably be: remove the cellId parameter from the createCell action & don't do anything in createCellReducer if the id already exists. Creating the cell object not-lazily is probably alright, as that structure is not stored on the server anyway (at least for now).

Thank you for taking the time to think this through! I fully agree with your conclusions and should have adapted the code accordingly.